xdsl/transforms/accfg/trace_acc_state.py

```diff
@@ -0,0 +1,74 @@
+"""
+These inference helpers are used to figure out which values are
+present at which points in the IR.
+
+Example inference results:
+
+```
+accfg.setup(A = %1, B = %2)  // previous state = {}
+// ...
+accfg.setup(A = %3, C = %1)  // previous state = {A = %1, B = %2}
+```
+
+These inference passes walk the IR backwards.
+"""
+
+from xdsl.dialects import accfg, scf
+from xdsl.ir import Block, SSAValue
+
+State = dict[str, SSAValue]
```

Comment:
I'm not sure what exactly it does but I have the impression that it helps Pyright a little bit if you add an annotation to say explicitly what's happening:
```suggestion
State: TypeAlias = dict[str, SSAValue]
```
Although, in this case, it feels like it might be worth adding some more structure and documentation to say what the object represents etc? Especially since `state_union` seems like it might as well be a method on the object. The lowest overhead way of doing that is this:
``` python
class AccState(NamedTuple):
    known_states: dict[str, SSAValue]
```
Like for `dataclass`, the init is auto generated

Reply:
@superlopuh I tend to agree, but I can not use the NamedTuple as you suggested (seems only available as factory method) and It doesn't seem to make the current code much more readable.

Reply:
I think you have to import it from `typing` for it to work this way, but I'm not sure, I've used it like this before and it's worked